### ISSUE TITLE
Have you heard of this similar extension by @pashaie?

### ISSUE THREAD

**Issue by devinrhode2** [ivanhofer/git-assistant#5]
It looks like its just for VIsual Studio proper. Also, inside vscode, when I search in the extensions pane for "submodule" or "submodules" I only see your extension. However, it may be interesting for you to look at:
https://github.com/pashaie/GitSubModulesVsPlugin

**Comment by ivanhofer** [comment]
No I haven't seen this extension before. Thanks for letting me know. I will take a closer look at it when I have more time.
I have planned a few things for the future (mostly bugfixing and speed improvements).
I'm always open for feedback and feature suggestions. Do you have something particular in mind? Maybe something from the extension you suggested that is missing here?

**Comment by devinrhode2** [comment]
It's pretty complicated but I'm looking at moving away from submodules to sibling repos, i.e.
```
clientA
clientB
shared-core
```
This is to discourage fragmenting code - I was reading TrunkBasedDevelopment.com and then found this talk from a very pregnant lady at google - https://youtu.be/W71BTkUbdqE

However, I can say that one thing which was annoying with this extension, it would be nice to be able to configure the warnings for being on master branch. Something your extension could help tremendously with, is remembering which branch a submodule is on, and automatically re-attaching it, IF the submodule sha exists on the branch you are trying to reattach to.

Some submodules default branch is not master, in which case we may want the warning, but overall it was not a warning that really helped at all.

**Comment by ivanhofer** [comment]
Yes I understand. Submodules are not always easy to work with. We use it a lot at work, because it fits our workflow, but they bring a lot of issues. That´s why I created the extension.

A monorepo-approach like you suggested is not always the best option. For projects that belong together (eg. an frontend, backend and api) that belong together a monorepo works well.

I´m not sure I understand your problem.
Do you mean the warning "you should not commit on this branch"? You can configure for which branches the warning should be displayed.
Or do you mean the automatic checkout of a branch when a detached HEAD is detectet?
The extension looks for the last commit ID on every branch and reattaches the HEAD if the commit IDs match. When two branches share the same commit ID it should appear QuickPick Panel so you can choose the correct branch.

Can you let me know if I understood your problem correctly or can you please write it down with other words?

Sorry for my late answer.
I went a few days offline ;)

**Comment by devinrhode2** [comment]
I did not know you can already configure which branches you get warnings for

your detached head solution sounds extremely good!

**Comment by ivanhofer** [comment]
The warnings will work for the main repository and all the submodules. You can edit them in the VS Code Settings Panel.
